Zend/zend_execute.c

```diff
@@ -621,17 +621,22 @@ static zend_never_inline ZEND_COLD void ZEND_FASTCALL zend_throw_non_object_erro
 	 || opline->opcode == ZEND_POST_INC_OBJ
 	 || opline->opcode == ZEND_POST_DEC_OBJ) {
 		zend_throw_error(NULL,
-			"Attempt to increment/decrement property '%s' of non-object",
-			ZSTR_VAL(property_name));
+			"Attempt to increment/decrement property '%s' on %s",
+			ZSTR_VAL(property_name), Z_ISUNDEF_P(object) ? "null" : zend_zval_type_name(object)
```

Comment:
I think it would be okay to adjust `zend_zval_type_name()` to handle this case directly, rather than having separate checks in so many places.

Reply:
I forgot to address this comment previously, so one of the 2 newly added fixups tried to do so.